### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by manuelgavidia**
This is connected to dojot/dojot#221

### CODE REVIEW

File: alarmlibrary/alarm.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+class Alarm(object):
+
+    def __init__(self, domain, severity, timestamp, namespace="", description=""):
+        if not isinstance(severity, AlarmSeverity):
+            raise ValueError('Invalid severity value, it must be AlarmSeverity')
+
+        self._domain = domain
+        self._severity = severity
+        self._timestamp = long(timestamp)
```
Comment (mmagr) [inline alarmlibrary/alarm.py]: If I'm reading this right, it seems we'll always be instantiating objects of this class on every alarm generation.
That being the case, why not have a default timestamp being set to `time.time()`?

File: alarmlibrary/alarm.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    WARNING = 0
+    MINOR = 1
+    MAJOR = 2
+    CRITICAL = 3
+    CLEAR = 4
+
+
+class Alarm(object):
+
+    def __init__(self, domain, severity, timestamp, namespace="", description=""):
```
Comment (mmagr) [inline alarmlibrary/alarm.py]: From the discussion on the information model of an alarm, it seems to me that namespace should always be sent, and is also used to identify (along with domain) an alarm.

If that's the case, this should be a mandatory field with no valid default shouldn't it?

File: alarmlibrary/alarm.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        self._domain = domain
+        self._severity = severity
+        self._timestamp = long(timestamp)
+        self._namespace = namespace
+        self._description = description
+
+        self._primary_subject = dict()
+        self._additional_data = dict()
+
+        severity_to_string = {
```
Comment (mmagr) [inline alarmlibrary/alarm.py]: shouldn't this be in `AlarmSeverity` ?

File: pylint.rc [hunk 4]
```diff
@@ -0,0 +1,377 @@
+[MASTER]
```
[the remark this answers is not in the document]
Comment (manuelgavidia) [reply pylint.rc]: Done.

File: examples/send_alarm.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    Parse command line parameters
+
+    :param args: command line parameters as list of strings
+    :return: command line parameters as :obj:`argparse.Namespace`
+    """
+    parser = argparse.ArgumentParser(
+        description="DOJOT Alarm publish example")
+    parser.add_argument(
+        dest="host",
+        help="Alarm manager host",
```
[the remark this answers is not in the document]
Comment (manuelgavidia) [reply examples/send_alarm.py]: Fixed

File: alarmlibrary/alarm.py (hunk 1 above)
Comment (manuelgavidia) [reply alarmlibrary/alarm.py]: Ok ... time.time() returns float. It will be converted to int

File: alarmlibrary/alarm.py (hunk 3 above)
Comment (manuelgavidia) [reply alarmlibrary/alarm.py]: Enum natively supports print, I willl change this

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+language: python
+install:
+  - pip install pylint
+script:
+  - pylint
```
Comment (mmagr) [inline .travis.yml]: perhaps (for default rc file - has more errors pending)?

```yaml
  - pylint alarmlibrary
```

or (has 2 pending errors)

```yaml
  - pylint alarmlibrary --rcfile=pylint.rc
```
https://travis-ci.org/mmagr/alarm-client-python/builds/346370545
(on next commit travis should run automatically)

**Pull request merged**